src/Nethermind/Nethermind.Synchronization/SnapSync/ProgressTracker.cs

```diff
@@ -471,5 +542,55 @@ public void Dispose()
                 range?.Dispose();
             }
         }
+
+        public interface IStorageRangeLock
+        {
+            IStorageRangeLock Increment(uint value = 1);
+            void Decrement(ValueHash256 key, bool removeFromOwner);
```

Comment:
Why `key` is needed here? Should it not be captured in the `ctor` of the lock, so that we just `Decrement(bool removeFromOwner)`. You get the lock by key so the key should not be needed here I think.